src/util/clipSketchToSubregion.ts

```diff
@@ -0,0 +1,227 @@
+    if (!geography) throw new Error("Specified geography doesn't exist");
+
+    const box = sketch.bbox || bbox(sketch);
+    const ds = project.getDatasourceById(
+      geography.datasourceId
+    ) as InternalVectorDatasource;
+    const subregion = await fgbFetchAll<Feature<Polygon | MultiPolygon>>(
+      project.dataBucketUrl() + getFlatGeobufFilename(ds),
+      box
+    );
+
+    if (!subregion[0]) {
+      console.log(
+        "Sketch/SketchCollection",
+        sketch.properties.name,
+        "has no overlap with geography",
+        geography.geographyId
+      );
+
+      const sketches = toSketchArray(sketch);
+      const finalsketches: Sketch<Polygon>[] = [];
+      sketches.forEach((sketch) => {
+        sketch.geometry = {
+          type: "Polygon",
+          coordinates: [[[0.0, 0.0]], [[0.0, 0.0]], [[0.0, 0.0]]],
+        };
+        finalsketches.push(sketch);
+      });
+
+      //Sketch
+      if (finalsketches.length === 1) {
+        return finalsketches[0];
+      }
+      //Sketch Collection
+      else {
+        return {
+          properties: sketch.properties,
+          bbox: box,
+          type: "FeatureCollection",
+          features: finalsketches,
+        };
+      }
+    } else {
+      const sketches = toSketchArray(sketch);
+      const finalsketches: Sketch<Polygon>[] = [];
+      sketches.forEach((sketch) => {
+        //const intersection = intersect(sketch.geometry, subregion[0].geometry);
+        const intersection = clipMultiMerge(
+          sketch,
+          featureCollection(subregion),
+          "intersection"
+        );
+        if (!intersection) console.log("no intersection");
+        intersection
+          ? (sketch.geometry = intersection.geometry as Polygon)
+          : (sketch.geometry = {
+              type: "Polygon",
+              coordinates: [[[0.0, 0.0]], [[0.0, 0.0]], [[0.0, 0.0]]],
+            });
+        finalsketches.push(sketch);
+      });
+
+      //Sketch
+      if (finalsketches.length === 1) {
+        return finalsketches[0];
+      }
+      //Sketch Collection
+      else {
+        return {
+          properties: sketch.properties,
+          bbox: box,
+          type: "FeatureCollection",
+          features: finalsketches,
+        };
+      }
+    }
+  }
+
+  return sketch;
+}
+
+/**
+ * Returns one aggregate object for every groupId present in metrics
+ * Each object includes following properties:
+ * numSketches - count of child sketches in the group
+ * [classId] - a percValue for each classId present in metrics for group
+ * value - sum of value across all classIds present in metrics for group
+ * percValue - given sum value across all classIds, contains ratio of total sum across all class IDs
+ */
+export async function clipSketchToSubregionSimple(
+  sketch: Sketch<Polygon> | SketchCollection<Polygon>,
+  extraParams: ExtraParams,
+  options: { tolerance: number; highQuality: boolean }
+): Promise<Sketch<Polygon> | SketchCollection<Polygon>> {
```

Comment:
Same as above, plus what differentiates this function from the above?  What makes it 'simple'?  If the caller has to choose between the two, what will help them?

Reply:
If the only difference is that one calls `simplify` at the end and the rest is duplicate code, then consider that simple version doesn't need to exist and the caller can call simplify on their own, or you can make a simple version, that just calls the base version and than adds a single simplify call before returning.

Reply:
There's some simplification happening internally, so I've edited the core function to accept optional simplification parameters. Let me know your thoughts on this as a solution.
https://github.com/seasketch/azores-nearshore-reports/blob/692a09dab82cfa76ea3c6956c32cbe74d3c5b25d/src/util/geography.ts#L67C1-L71C58

Reply:
Sounds good.  Another function we can flag to contribute back with non-breaking improvements.